client_app/src/components/shelter/PastVolunteers.js

```diff
@@ -1,22 +1,30 @@
-import React, { Component } from "react";
 import AccountCircleIcon from "@mui/icons-material/AccountCircle";
 import NotificationsNoneIcon from "@mui/icons-material/NotificationsNone";
 import "../../styles/shelter/PastVolunteers.css";
 
-class PastVolunteers extends Component {
-  render() {
-    const { shiftDetails } = this.props;
+const PastVolunteers = ({shiftDetails}) => {
+    const currentTime = Date.now();
+    const filteredShifts = shiftDetails.filter(item => item.start_time < currentTime);
+    const workerList = filteredShifts
+    .flatMap(item => item.worker ? item.worker.split(",").map(name => name.trim()) : []);
+    const uniqueWorker = [...new Set(workerList)];
 
     return (
-      <div className="volunteers-item-container">
-        <div className="volunteers-item">
-          <AccountCircleIcon />
-          <span>{shiftDetails.label}</span>
-          <NotificationsNoneIcon />
-        </div>
+      <div className="past-volunteers-container">
+        {uniqueWorker.length > 0 ? (uniqueWorker.slice(0,3).map((name, index) => (
```

Comment:
Is this hardcoded just for now? If not we might need to expand this eventually ( where it limits the users that appear to 3) depending on how many volunteers a shelter requires.

Reply:
We only show three past volunteers on the Shelter Dashboard. Since this is a dashboard, we should only display a few.